File: PHP/Structural/Adapter/example/Adapter.php
<?php

namespace TheRayCode\Adapter;

class Adapter implements Target

{
    private $adaptee;

    public function __construct(Adaptee $adaptee)
    {
        $this->adaptee = $adaptee;
    }

    public function request(): string
    {
        return "Adapter: " . $this->adaptee->specificRequest();
    }
}

File: CPP/Structural/Adapter/Example/Adapter.cpp
//
// Created by ray on 8/29/23.
//
#include "Adapter.h"
#include <iostream>

void Adapter::Request() {
    std::cout << "[Adapter] Calling SpecificRequest\n" ;
    //m_Adaptee.SpecificRequest() ;
    SpecificRequest() ;
}